### ISSUE TITLE
Absolutify base href

### ISSUE THREAD

**Issue by jaimeiniesta** [jaimeiniesta/metainspector#240]
Some pages like https://www.delta.com/us/en have a relative base href tag:

```html
<base href="/">
```

This makes the scraping fail because we expect it to be an absolute URL.

To fix this, we should also absolutify this base href with the url of the scraped page. If the base href was already an absolute one, it won't get changed.

**Comment by navarasu** [comment]
We can handle this similar to blank base href.

**Comment by jaimeiniesta** [comment]
No, I don't think a base href of "/" should be treated as an empty one. It means different things: if empty, it need to be ignored, but if it says "/", the document author is trying to say that relative links should be built from the **root** directory. For example:

Let's say there's a page http://example.com/some/dir/first.html and it has a link:

```html
<a href="second.html">Second page</a>
```

When there is no base href (or it is empty and we ignore it), this relative link will be absolutified as http://example.com/some/dir/second.html

Instead, if the base href is `/` it should be treated as if it was `http://example.com/` so the absolutified link would be http://example.com/second.html

If the base href was `/other` then the absolutified link would be http://example.com/other/second.html

**Comment by navarasu** [comment]
Yeah. It makes sense.

**Comment by navarasu** [comment]
```

**Comment by navarasu** [comment]
@jaimeiniesta  Please check this PR. Time being I have overridden this method in my project to fix the failure.
